File: tests/com/jetbrains/idear/ActionRecognizerTest.java
package com.jetbrains.idear;

import com.jetbrains.idear.actions.recognition.ActionCallInfo;
import com.jetbrains.idear.actions.recognition.ExtractActionRecognizer;
import org.junit.Test;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertTrue;

public class ActionRecognizerTest {

    @Test
    public void test_extract_with_name() throws Exception {
        String text = "idea extract to variable myTest";

        ExtractActionRecognizer recognizer = new ExtractActionRecognizer();
        assertTrue(recognizer.isMatching(text));

        ActionCallInfo info = recognizer.getActionInfo(text);
        assertEquals("IntroduceVariable", info.actionId);
        assertEquals("myTest", info.typeAfter);
    }

    @Test
    public void test_extract_without_name() throws Exception {
        String text = "idea extract to variable";
        ExtractActionRecognizer recognizer = new ExtractActionRecognizer();
        assertTrue(recognizer.isMatching(text));

        ActionCallInfo info = recognizer.getActionInfo(text);
        assertEquals("IntroduceVariable", info.actionId);
    }
}

File: src/test/kotlin/org.openasr.idear/ActionRecognizerTest.kt
package org.openasr.idear

import com.intellij.openapi.actionSystem.DataContext
import com.intellij.openapi.actionSystem.DataKey
import com.intellij.openapi.actionSystem.impl.SimpleDataContext
import org.junit.Assert.*
import org.junit.Test
import org.openasr.idear.actions.recognition.ExtractActionRecognizer

class ActionRecognizerTest {
    private val emptyDataContext: DataContext
        get() = SimpleDataContext.getSimpleContext(DataKey.create(""), Any())

    @Test
    fun test_extract_with_name() {
        val text = "idea extract to variable myTest"

        val recognizer = ExtractActionRecognizer()
        assertTrue(recognizer.isMatching(text))

        val info = recognizer.getActionInfo(text, emptyDataContext)
        assertEquals("IntroduceVariable", info!!.actionId)
        assertEquals("myTest", info.typeAfter)
    }

    @Test
    fun test_extract_without_name() {
        val text = "idea extract to variable"
        val recognizer = ExtractActionRecognizer()
        assertTrue(recognizer.isMatching(text))

        val info = recognizer.getActionInfo(text, emptyDataContext)
        assertEquals("IntroduceVariable", info!!.actionId)
    }
}
